paasta_tools/setup_prometheus_adapter_config.py

```diff
@@ -288,6 +304,96 @@ def create_instance_uwsgi_scaling_rule(
+    paasta_cluster: str,
+) -> PrometheusAdapterRule:
+    """
+    Creates a Prometheus adapter rule config for a given service instance.
+    """
+    setpoint = autoscaling_config["setpoint"]
+    moving_average_window = autoscaling_config.get(
+        "moving_average_window_seconds", DEFAULT_UWSGI_AUTOSCALING_MOVING_AVERAGE_WINDOW
+    )
+    # this should always be set, but we default to 0 for safety as the worst thing that would happen
+    # is that we take a couple more iterations than required to hit the desired setpoint
+    offset = autoscaling_config.get("offset", 0)
+    deployment_name = get_kubernetes_app_name(service=service, instance=instance)
+    worker_filter_terms = f"paasta_cluster='{paasta_cluster}',paasta_service='{service}',paasta_instance='{instance}'"
+    replica_filter_terms = (
+        f"paasta_cluster='{paasta_cluster}',deployment='{deployment_name}'"
+    )
+
+    current_replicas = f"""
+        sum(
+            label_join(
+                (
+                    kube_deployment_spec_replicas{{{replica_filter_terms}}} >= 0
+                    or
+                    max_over_time(
+                        kube_deployment_spec_replicas{{{replica_filter_terms}}}[{DEFAULT_EXTRAPOLATION_TIME}s]
+                    )
+                ),
+                "kube_deployment", "", "deployment"
+            )
+        ) by (kube_deployment)
+    """
+    # k8s:deployment:pods_status_ready is a metric created by summing kube_pod_status_ready
+    # over paasta service/instance/cluster. it counts the number of ready pods in a paasta
+    # deployment.
+    ready_pods = f"""
+        (sum(
+            k8s:deployment:pods_status_ready{{{worker_filter_terms}}} >= 0
+            or
+            max_over_time(
+                k8s:deployment:pods_status_ready{{{worker_filter_terms}}}[{DEFAULT_EXTRAPOLATION_TIME}s]
+            )
+        ) by (kube_deployment))
+    """
+    load_per_instance = f"""
+        (piscina_queue_size{{{worker_filter_terms}}} / piscina_max_queue_size{{{worker_filter_terms}}}) by (kube_pod, kube_deployment)
+    """
+    missing_instances = f"""
+        clamp_min(
+            {ready_pods} - count({load_per_instance}) by (kube_deployment),
+            0
+        )
+    """
+    total_load = f"""
+    (
+        sum(
+            {load_per_instance}
+        ) by (kube_deployment)
+        +
+        {missing_instances}
+    )
+    """
+    desired_instances_at_each_point_in_time = f"""
+        {total_load} / {setpoint - offset}
+    """
+    desired_instances = f"""
+        avg_over_time(
+            (
+                {desired_instances_at_each_point_in_time}
+            )[{moving_average_window}s:]
+        )
+    """
+    metrics_query = f"""
+        {desired_instances} / {current_replicas}
+    """
+
+    metric_name = f"{deployment_name}-uwsgi-prom"
+
+    return {
+        "name": {"as": metric_name},
+        "seriesQuery": f"piscina_queue_load{{{worker_filter_terms}}}",
```

Comment:
unsure if this name is correct

Reply:
this needs to be a query that will return data for the resources that will be autoscaled - so 
```suggestion
        "seriesQuery": f"piscina_queue_size{{{worker_filter_terms}}}",
```
or
```suggestion
        "seriesQuery": f"piscina_max_queue_size{{{worker_filter_terms}}}",
```
are probably good enough options (in case you're curious: https://github.com/kubernetes-sigs/prometheus-adapter/blob/master/docs/config.md if you want to read up on what this config is being used for in the end)